src/net/SDP/SDP.cs

```diff
@@ -309,13 +309,27 @@ public static SDP ParseSDPDescription(string sdpDescription)
                                 break;
 
                             case var l when l.StartsWith("m="):
-                                Match mediaMatch = Regex.Match(sdpLineTrimmed.Substring(2), @"(?<type>\w+)\s+(?<port>\d+)\s+(?<transport>\S+)(\s*)(?<formats>.*)$");
+                                Match mediaMatch = Regex.Match(
+                                        sdpLineTrimmed.Substring(2),
+                                        @"(?<type>\w+)\s+(?<port>\d+)(?:\/(?<portCount>\d+))?\s+(?<transport>\S+)\s*(?<formats>.*)$"
+                                    );

```

Comment:
This regex has the potential to be parsed every time it's used.

Consider using [.NET regular expression source generators](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators).

A regex here might be overkill. Have you considered using [StringTokenizer Struct](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.primitives.stringtokenizer)?

Reply:
no i just made a small update to what was already there. if you have a better way of doing it please feel free to modify